dyne/examples.py

```diff
@@ -533,3 +534,158 @@ def h(k, X, with_jacobian=True):
 
     return NonlinearProblemExample(X0, P0, f, Q, [(measurement_epochs, Z, h, R)],
                                    n_epochs, Xt, Wt)
+
+
+def generate_magnetic_heading(
+    total_time=120,
+    time_step=0.1,
+    rph_mean=[3, -5, 0],
+    rph_change_amplitude=5,
+    rph_change_period=[5, 5, 30],
+    rph_change_phase_offset=[0, 90, 45],
+    magnetic_field_n=[10, 3, 40],
+    mag_bias=[30, -20, 70],
+    X0=None,
+    P0=np.diag([180, 50, 50, 50])**2,
+    q=np.zeros(4),
+    measurement_sd=[1, 1, 1],
+    rng=0
+):
+    """Generate data for exapmle of heading estimation using biased magnetometer data.
+
+    Roll, pitch and heading angles are varying according to a given harmonic law.
+    Exact values of roll and pitch are known, heading and magnetometer biases
+    should be estimated.
+
+    The discrete time system model is::
+
+        h[k+1] = h[k] + dh[k] + w_h[k]
+        bx[k+1] = bx[k] + w_x[k]
+        by[k+1] = by[k] + w_y[k]
+        bz[k+1] = bz[k] + w_z[k]
+
+    Where::
+
+        h  - heading
+        dh - known heading increment, used as control
+        bx, by, bz - magnetometer biases, components of `mag_bias`
+        wx, wy, wz - discrete noise
+
+    The measurement model is::
+
+        [mx, my, mz] = C_bn @ [mn, me, md] + [bx, by, bz]
+
+    Where::
+
+        C_bn - rotation matrix from ``body`` to ``world`` frame
+        mn, me, md - magnetic field in ``world`` frame, componets of `magnetic_field_n`
+        bx, by, bz - magnetometer biases, components of `mag_bias`
+        mx, my, mz - magnetometer measurements
+
+    Parameters
+    ----------
+    total_time : float
+        Total time of the simulation.
+    time_step : float
+        Discretization time step.
+    rph_mean: array_like, shape (3,)
+        Mean value of angles in degrees.
+    rph_change_amplitude: float or array_like, shape (3,)
+        Amplitude of angles variation in degrees.
+    rph_change_period: float or array_like, shape (3,)
+        Period of sinusoidal angles change in seconds.
+    rph_change_phase_offset: array_like, shape (3,)
+        Phase offset for sinusoid part in degrees.
+    magnetic_field_n: array_like, shape (3,)
+        Magnetic field in ``world`` frame.
```

Comment:
What do double ticks mean here?

Not sure if "n" will make sense for other readers. Possibly need to explain it, like "Components of magnetic field expressed in the world frame (denoted as n)".

Reply:
Agree